nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/ValidateCsv.java

```diff
@@ -0,0 +1,613 @@
+                    logger.error("Failed to validate {} against schema due to {}", new Object[]{flowFile}, e);
+                } finally {
+                    if(listReader != null) {
+                        listReader.close();
+                    }
+                }
+            }
+        });
+
+        if(isWholeFFValidation) {
+            if (valid.get()) {
+                logger.info("Successfully validated {} against schema; routing to 'valid'", new Object[]{flowFile});
+                session.getProvenanceReporter().route(flowFile, REL_VALID);
+                session.transfer(flowFile, REL_VALID);
+            } else {
+                session.getProvenanceReporter().route(flowFile, REL_INVALID);
+                session.transfer(flowFile, REL_INVALID);
+            }
+        } else {
+            if (valid.get()) {
+                logger.debug("Successfully validated {} against schema; routing to 'valid'", new Object[]{validFF.get()});
+                session.getProvenanceReporter().route(validFF.get(), REL_VALID, "All " + totalCount.get() + " line(s) are valid");
+                session.transfer(validFF.get(), REL_VALID);
+                session.remove(invalidFF.get());
+                session.remove(flowFile);
+            } else if (okCount.get() != 0) {
+                if(header) {
+                    totalCount.set(totalCount.get() - 1);
+                }
+
+                logger.debug("Successfully validated {}/{} line(s) in {} against schema; routing valid lines to 'valid' and invalid lines to 'invalid'",
+                        new Object[]{okCount.get(), totalCount.get(), flowFile});
+                session.getProvenanceReporter().route(validFF.get(), REL_VALID, okCount.get() + " valid line(s)");
+                session.transfer(validFF.get(), REL_VALID);
+                session.getProvenanceReporter().route(invalidFF.get(), REL_INVALID, (totalCount.get() - okCount.get()) + " invalid line(s)");
+                session.transfer(invalidFF.get(), REL_INVALID);
+                session.remove(flowFile);
+            } else {
+                logger.debug("All lines in {} are invalid; routing to 'invalid'", new Object[]{invalidFF.get()});
+                session.getProvenanceReporter().route(invalidFF.get(), REL_INVALID, "All " + totalCount.get() + " line(s) are invalid");
+                session.transfer(invalidFF.get(), REL_INVALID);
+                session.remove(validFF.get());
+                session.remove(flowFile);
+            }
+        }
+    }
+
+    /**
+     * Method used to correctly write the lines by taking into account end of line
+     * character and separator character.
+     * @param list list of elements of the current row
+     * @param csvPref CSV preferences
+     * @param isFirstLine true if this is the first line we append
+     * @return String to append in the flow file
+     */
+    private byte[] print(List<?> list, CsvPreference csvPref, boolean isFirstLine) {
```

Comment:
Not sure why but after being routed through the processor instead of '\n' the new line character the literals "\" and "n" were added. Below are two screen shots of the data below and after (only line routed to invalid)

![screen shot 2016-07-27 at 7 23 52 pm](https://cloud.githubusercontent.com/assets/11302527/17196060/e4267118-542f-11e6-8831-83f8389a615a.png)
![screen shot 2016-07-27 at 7 23 55 pm](https://cloud.githubusercontent.com/assets/11302527/17196059/e424d600-542f-11e6-8e35-5f1274912ef1.png)